src/dataworker/Dataworker.ts

```diff
@@ -2719,6 +2724,80 @@ export class Dataworker {
     return l2MessagerContract.minimumFeeInWei();
   }
 
+  /**
+   * Compute the msg.value to attach when executing a relayer refund leaf on L2.
+   * - If amountToReturn == 0 or non-EVM chain: undefined
+   * - If OFT is configured on the SpokePool for the token: quote native fee via OFT messenger and return fee*2
+   * - Else if Linea: return the pre-fetched Linea L2->L1 message fee
+   * - Additionally, enforce OFT-on-Linea mutual exclusion and throw if encountered
+   */
+  private async _getMsgValueForRelayerRefundLeaf(
+    client: SpokePoolClient,
+    leaf: RelayerRefundLeaf
+  ): Promise<BigNumber | undefined> {
+    // We don't support providing SOL value with the executor call, so if spokePool is not evm, or the return amount is 0, exit early
+    if (!isEVMSpokePoolClient(client) || !leaf.amountToReturn.gt(0)) {
+      return undefined;
+    }
+
+    // If the chain is Linea, then we need to allocate ETH in the call to executeRelayerRefundLeaf
+    const lineaMsgValuePortion = sdkUtils.chainIsLinea(client.chainId)
+      ? await this._getRequiredEthForLineaRelayLeafExecution(client)
+      : BigNumber.from(0);
```

Comment:
nit (removes an extra memory allocation)
```suggestion
      : bnZero;
```

Reply:
[57038ab](https://github.com/across-protocol/relayer/pull/2528/commits/57038abfea2ae749df35ba73eeb3dbb853807903)